tests/test_fairworkflow.py

```diff
@@ -14,17 +14,39 @@
 class TestFairWorkflow:
```

Comment:
Sorry for all the unrelated changes regarding fixtures here... I had to fix this to make the tests not mutate the objects (making the test success dependent on which tests are run before it).

Reply:
Only the last two tests (scroll down) were actually changed.

Reply:
Clearly this is totally unacceptable behaviour, Sven, but I will let it pass this one time.

Reply:
Lol